crates/pallets/starknet/src/simulations.rs

```diff
@@ -68,7 +90,12 @@ impl<T: Config> Pallet<T> {
 
         let mut fees = Vec::with_capacity(transactions_len);
         for fee_res in fee_res_iterator {
-            fees.push(fee_res??);
+            match fee_res {
+                Ok(fee) => fees.push(fee?),
+                Err(Error::<T>::TransactionExecutionFailed) => fees.push(FEE_ESTIMATION_FAILED),
+                Err(Error::<T>::TransactionExecutionReverted) => fees.push(FEE_ESTIMATION_REVERTED),
+                Err(e) => return Err(e.into()),
+            }
```

Comment:
https://github.com/starkware-libs/starknet-specs/blob/76bdde23c7dae370a3340e40f7ca2ef2520e75b9/api/starknet_api_openrpc.json#L614

For estimate fee, we abort as soon as the first estimate fail.
That is why the previous code was `fees.push(fee_res??);`, where `?` immediately returns.
You should keep this behaviour of early exiting the loop at the first error encountered